### ISSUE TITLE
Export (v)methods to lua

### ISSUE THREAD

**Issue by warmist** [DFHack/dfhack#1542]
Just a random idea - https://github.com/DFHack/dfhack/blob/0.47.04-beta1/library/LuaTypes.cpp#L547 maybe adding stuff here would allow to get more info about vmethods/functions from lua?

**Comment by BenLubar** [comment]
You can call them, but there's not a way to ask for them as a field reference.

**Comment by warmist** [comment]
> You can call them, but there's not a way to ask for them as a field reference.

Yup. Also ipairs like thing (vpairs?) that would list all of functions. This would greatly simplify scripting as you could lookup stuff faster instead of using xmls or headers.

**Comment by lethosor** [comment]
I'm not sure how much useful information the function metatables actually have, but worth investigating. Note that ipairs is a Lua built-in, like pairs, and uses the `__ipairs` metamethod (I think). Implementing a third variant with the same functionality could require patching Lua, which isn't ideal.

**Comment by warmist** [comment]
Ah i meant something like ipairs i.e. we don't need to implement any metamethods and usage could be as simple as `for name, vmethod in vmethods(obj or obj.identity) do...`

TBH i'm not that familiar with details to know how hard it would be to implement.
